pkg/approvaltest/approvals.go

```diff
@@ -115,22 +111,42 @@ func approveEventDocs(t testing.TB, name string, eventDocs [][]byte, dynamic ...
 		"observer.version",
 	}, dynamic...)
 
-	// Rewrite all dynamic fields to have a known value,
-	// so dynamic fields don't affect diffs.
-	events := make([]interface{}, len(eventDocs))
-	for i, doc := range eventDocs {
+	for i := range srcs {
 		for _, field := range dynamic {
-			existing := gjson.GetBytes(doc, field)
+			field = strings.ReplaceAll(field, ".", "\\.")
```

Comment:
Is this going to work for `_source` as well as `fields`? We might need to switch based on whether it's ApproveEvents vs. ApproveFields

Reply:
@axw You are right, it wouldn't have worked with `ApproveEvents`, I have fixed it now. Thanks for catching this.